---
title: InvalidRpcUrlError
description: Represents an error that occurs when an invalid RPC URL is provided. This class extends the `BaseError` class and includes the invalid URL in the error message.
slug: wallets/reference/aa-sdk/core/src/classes/InvalidRpcUrlError
layout: reference
---

{/* This file is auto-generated by TypeDoc. Do not edit manually. */}

Defined in: [aa-sdk/core/src/errors/client.ts:30](https://github.com/alchemyplatform/aa-sdk/blob/9f926c402d1661309abbf8eae8431e52e8c5e53f/aa-sdk/core/src/errors/client.ts#L30)

Represents an error that occurs when an invalid RPC URL is provided. This class extends the `BaseError` class and includes the invalid URL in the error message.

## Extends

- [`BaseError`](BaseError.mdx)

## Constructors

### Constructor

```ts
new InvalidRpcUrlError(rpcUrl?): InvalidRpcUrlError;
```

Defined in: [aa-sdk/core/src/errors/client.ts:38](https://github.com/alchemyplatform/aa-sdk/blob/9f926c402d1661309abbf8eae8431e52e8c5e53f/aa-sdk/core/src/errors/client.ts#L38)

Creates an instance of an error with a message indicating an invalid RPC URL.

#### Parameters

<table>
  <thead>
    <tr>
      <th align="left">Parameter</th>
      <th align="left">Type</th>
      <th align="left">Description</th>
    </tr>
  </thead>

  <tbody>
    <tr>
      <td>
        `rpcUrl?`
      </td>

      <td>
        `string`
      </td>

      <td>
        The invalid RPC URL that caused the error
      </td>
    </tr>

  </tbody>
</table>

#### Returns

`InvalidRpcUrlError`

#### Overrides

[`BaseError`](BaseError.mdx).[`constructor`](BaseError.mdx#constructor)

## Properties

<table>
  <thead>
    <tr>
      <th align="left">Property</th>
      <th align="left">Type</th>
      <th align="left">Default value</th>
      <th align="left">Description</th>
    </tr>
  </thead>

  <tbody>
    <tr>
      <td>
        <a id="cause" /> `cause?`
      </td>

      <td>
        `unknown`
      </td>

      <td>
        `undefined`
      </td>

      <td>
        ‐
      </td>
    </tr>

    <tr>
      <td>
        <a id="details" /> `details`
      </td>

      <td>
        `string`
      </td>

      <td>
        `undefined`
      </td>

      <td>
        ‐
      </td>
    </tr>

    <tr>
      <td>
        <a id="docspath" /> `docsPath?`
      </td>

      <td>
        `string`
      </td>

      <td>
        `undefined`
      </td>

      <td>
        ‐
      </td>
    </tr>

    <tr>
      <td>
        <a id="message" /> `message`
      </td>

      <td>
        `string`
      </td>

      <td>
        `undefined`
      </td>

      <td>
        ‐
      </td>
    </tr>

    <tr>
      <td>
        <a id="metamessages" /> `metaMessages?`
      </td>

      <td>
        `string`\[]
      </td>

      <td>
        `undefined`
      </td>

      <td>
        ‐
      </td>
    </tr>

    <tr>
      <td>
        <a id="name" /> `name`
      </td>

      <td>
        `string`
      </td>

      <td>
        `"InvalidRpcUrlError"`
      </td>

      <td>
        ‐
      </td>
    </tr>

    <tr>
      <td>
        <a id="shortmessage" /> `shortMessage`
      </td>

      <td>
        `string`
      </td>

      <td>
        `undefined`
      </td>

      <td>
        ‐
      </td>
    </tr>

    <tr>
      <td>
        <a id="stack" /> `stack?`
      </td>

      <td>
        `string`
      </td>

      <td>
        `undefined`
      </td>

      <td>
        ‐
      </td>
    </tr>

    <tr>
      <td>
        <a id="version" /> `version`
      </td>

      <td>
        `string`
      </td>

      <td>
        `VERSION`
      </td>

      <td>
        ‐
      </td>
    </tr>

    <tr>
      <td>
        <a id="stacktracelimit" /> `stackTraceLimit`
      </td>

      <td>
        `number`
      </td>

      <td>
        `undefined`
      </td>

      <td>
        The `Error.stackTraceLimit` property specifies the number of stack frames
        collected by a stack trace (whether generated by `new Error().stack` or
        `Error.captureStackTrace(obj)`).

        The default value is `10` but may be set to any valid JavaScript number. Changes
        will affect any stack trace captured *after* the value has been changed.

        If set to a non-number value, or set to a negative number, stack traces will
        not capture any frames.
      </td>
    </tr>

  </tbody>
</table>

## Methods

### walk()

#### Call Signature

```ts
walk(): Error;
```

Defined in: node_modules/viem/\_types/errors/base.d.ts:26

##### Returns

`Error`

##### Inherited from

[`BaseError`](BaseError.mdx).[`walk`](BaseError.mdx#walk)

#### Call Signature

```ts
walk(fn): null | Error;
```

Defined in: node_modules/viem/\_types/errors/base.d.ts:27

##### Parameters

<table>
  <thead>
    <tr>
      <th align="left">Parameter</th>
      <th align="left">Type</th>
    </tr>
  </thead>

  <tbody>
    <tr>
      <td>
        `fn`
      </td>

      <td>
        (`err`) => `boolean`
      </td>
    </tr>

  </tbody>
</table>

##### Returns

`null` | `Error`

##### Inherited from

[`BaseError`](BaseError.mdx).[`walk`](BaseError.mdx#walk)

---

### captureStackTrace()

```ts
static captureStackTrace(targetObject, constructorOpt?): void;
```

Defined in: node_modules/@types/node/globals.d.ts:145

Creates a `.stack` property on `targetObject`, which when accessed returns
a string representing the location in the code at which
`Error.captureStackTrace()` was called.

```js
const myObject = {};
Error.captureStackTrace(myObject);
myObject.stack; // Similar to `new Error().stack`
```

The first line of the trace will be prefixed with
`${myObject.name}: ${myObject.message}`.

The optional `constructorOpt` argument accepts a function. If given, all frames
above `constructorOpt`, including `constructorOpt`, will be omitted from the
generated stack trace.

The `constructorOpt` argument is useful for hiding implementation
details of error generation from the user. For instance:

```js
function a() {
  b();
}

function b() {
  c();
}

function c() {
  // Create an error without stack trace to avoid calculating the stack trace twice.
  const { stackTraceLimit } = Error;
  Error.stackTraceLimit = 0;
  const error = new Error();
  Error.stackTraceLimit = stackTraceLimit;

  // Capture the stack trace above function b
  Error.captureStackTrace(error, b); // Neither function c, nor b is included in the stack trace
  throw error;
}

a();
```

#### Parameters

<table>
  <thead>
    <tr>
      <th align="left">Parameter</th>
      <th align="left">Type</th>
    </tr>
  </thead>

  <tbody>
    <tr>
      <td>
        `targetObject`
      </td>

      <td>
        `object`
      </td>
    </tr>

    <tr>
      <td>
        `constructorOpt?`
      </td>

      <td>
        `Function`
      </td>
    </tr>

  </tbody>
</table>

#### Returns

`void`

#### Inherited from

[`BaseError`](BaseError.mdx).[`captureStackTrace`](BaseError.mdx#capturestacktrace)

---

### prepareStackTrace()

```ts
static prepareStackTrace(err, stackTraces): any;
```

Defined in: node_modules/@types/node/globals.d.ts:149

#### Parameters

<table>
  <thead>
    <tr>
      <th align="left">Parameter</th>
      <th align="left">Type</th>
    </tr>
  </thead>

  <tbody>
    <tr>
      <td>
        `err`
      </td>

      <td>
        `Error`
      </td>
    </tr>

    <tr>
      <td>
        `stackTraces`
      </td>

      <td>
        `CallSite`\[]
      </td>
    </tr>

  </tbody>
</table>

#### Returns

`any`

#### See

https://v8.dev/docs/stack-trace-api#customizing-stack-traces

#### Inherited from

[`BaseError`](BaseError.mdx).[`prepareStackTrace`](BaseError.mdx#preparestacktrace)
